### PR TITLE
Enhancement#01 - Change number of products added to cart

### PR DESCRIPTION
**Pull request by rihan-a**
## Description
An upgrade to the user experience, aiming at the ease of use 
Users can choose how many products they would like to add to cart from both Home page & Shop page

## Screenshots
Screenshot 2022-09-02 at 14 58 05

## Testing
1- Open website Home page or Shop page
2- Go to a product
3- Press (+) to add product to cart and (-) to remove product from cart
4- Go to to cart page to make sure the added products are there 
4- Cart count should be updated in all pages

### CODE REVIEW

File: assets/css/style.css [hunk 1]
```diff
@@ -70,10 +70,34 @@ body {
   padding: 0 40px;
 }
 
+/* --- Add to cart Button + / -   --- */
+
+.pro-minus,
```
Comment (SabryShawally) [inline assets/css/style.css:75]: What does ‘pro-‘ stand for? I find it difficult to know what it references/means just by reading.

File: assets/css/style.css [hunk 2]
```diff
@@ -70,10 +70,34 @@ body {
...
+  color: black;
+  background-color: white;
+}
+
 @media (max-width: 799px) {
   .container {
     padding: 0 20px;
   }
+
+  .pro-minus,
```
Comment (SabryShawally) [inline assets/css/style.css]: Similarly here

File: assets/css/style.css [hunk 3]
```diff
@@ -297,18 +321,16 @@ body {
   }
 }
 
-/* --------------- Featured products ---------------  /  ----------------------------------- Shop Page ---------------------------------- */
+/* ---------- Featured products ----------  / -------------- Shop Page --------------------- */
```
Comment (SabryShawally) [inline assets/css/style.css:324]: Why not separate to different css files for each page? This way you minimize the data that needs to be transferred for each page and only load what you need for the page displayed

File: assets/css/style.css [hunk 4]
```diff
@@ -369,21 +391,12 @@ body {
   height: 40px;
 }
 
-#product1 .pro .desc h4:hover {
```
Comment (SabryShawally) [inline assets/css/style.css:372]: What is product 1? That is difficult to understand what it means

File: assets/css/style.css [hunk 5]
```diff
@@ -544,9 +585,18 @@ body {
...
+  .price-cart-container {
+    flex-direction: column;
+  }
+
   #product1 .pro .desc h4 {
     height: 50px;
     font-size: 0.9rem;
+    text-align: center;
+    margin-bottom: 10px;
+  }
```
Comment (SabryShawally) [inline assets/css/style.css]: Please add proper spacing between the styles

File: assets/scripts/cart.js [hunk 6]
```diff
@@ -120,12 +124,15 @@ function changeNumberOfUnits(action, id) {
 
 // Function to clear the cart from products
 function clearCart() {
-  let emptyCart = [];
   cartContainer.innerHTML = "";
   cartSubtotal.innerHTML = "EGP 0";
   cartCount.innerHTML = `Cart(0)`;
-  localStorage.setItem("cartCount", 0);
-  localStorage.setItem("cart", JSON.stringify(emptyCart));
+  cartCount2.innerHTML = `Cart(0)`;
```
Comment (SabryShawally) [inline assets/scripts/cart.js]: What is cartCount2 ? Please use meaningful names

File: assets/scripts/fetchProducts.js [hunk 7]
```diff
@@ -8,5 +9,22 @@ export async function getProducts() {
     console.log(error);
   }
 }
-
 getProducts();
+
+export async function saveProductsToLocal() {
```
Comment (SabryShawally) [inline assets/scripts/fetchProducts.js:14]: The name of the file doesn’t seem to match what this function does. I expect it to only do read only changes from the words fetchProducts. But this method also saves to local store

File: assets/scripts/fetchProducts.js [hunk 8]
```diff
@@ -8,5 +9,22 @@ export async function getProducts() {
...
+    let modifiedProductsList = [];
+    allProducts.products.forEach(function (item) {
+      if (item.vendor === "rihan.") {
+        modifiedProductsList.push({
+          ...item,
+          numberOfUnits: 0,
+        });
+      }
+    });
+    localStorage.setItem("allProducts", JSON.stringify(modifiedProductsList));
```
Comment (SabryShawally) [inline assets/scripts/fetchProducts.js:27]: Why not just save it as a json object?

File: assets/scripts/global.js [hunk 9]
```diff
@@ -10,9 +10,9 @@ let menuLeft = document.querySelector(".header-left");
...
 
 // Testing the touch events to fix the links bug on IOS
-hamburgerMenu.addEventListener("touchstart", showHideMenu, false);
+hamburgerMenu.addEventListener("click", showHideMenu, false);
 // hamburgerMenu.addEventListener("touchmove", showHideMenu, false);
-// hamburgerMenu.addEventListener("touchend", showHideMenu, false);
+hamburgerMenu.addEventListener("click", showHideMenu, false);
 
 // function to show/hide the hamburger menu on small screens
 function showHideMenu() {
```
Comment (SabryShawally) [inline assets/scripts/global.js:18]: The name is confusing I would suggest toggleMenu

File: assets/scripts/home.js [hunk 10]
```diff
@@ -1,48 +1,58 @@
...
+
+import { changeNumberOfUnits } from "./changeNumber.js";
+import { updateCartCount } from "./updateCartCount.js";
+import { saveProductsToLocal } from "./fetchProducts.js";
 
 // Variables
 let featuredProductsContainer = document.querySelector(
   ".featured-pro-container"
 );
+saveProductsToLocal();
```
Comment (SabryShawally) [inline assets/scripts/home.js:12]: This says save, but it then does print? I would also expect it to take an argument of what it’s meant to save

File: assets/scripts/home.js [hunk 11]
```diff
@@ -1,48 +1,58 @@
...
+        <button class="pro-plus" type="button"> <a class="product-plus" productID="${allProducts[i].id}" > + </a> </button>
         </div>
+</div>
       </div>
       </div>`;
     }
-  } else {
-    alert("No data available.");
   }
+  // else {
```
Comment (SabryShawally) [inline assets/scripts/home.js]: Please remove unused code

File: assets/scripts/home.js [hunk 12]
```diff
@@ -51,13 +61,46 @@ function openProductPage(e) {
     e.target.classList.contains("product-img")
   ) {
     let id = e.target.getAttribute("productID");
-    openProduct(id);
+    setTimeout(() => {
+      window.location = "product.html";
+    }, 100);
+    console.log(id);
+    localStorage.setItem("productID", id);
```
Comment (SabryShawally) [inline assets/scripts/home.js]: Should you not save before redirecting?

File: assets/scripts/home.js [hunk 13]
```diff
@@ -51,13 +61,46 @@ function openProductPage(e) {
...
-    addToCart(id);
+    //removeFromCart(id);
+    changeNumberOfUnits("minus", id);
+    featuredProductsContainer.innerHTML = "";
+    printFeaturedProducts();
+    updateCartCount();
   }
 }
+
+// Touch event for + / - styling
```
Comment (SabryShawally) [inline assets/scripts/home.js]: ditto

File: assets/scripts/newCart.js [hunk 14]
```diff
@@ -0,0 +1,134 @@
+// Imports
+import { changeNumberOfUnits } from "./changeNumber.js";
```
Comment (SabryShawally) [inline assets/scripts/newCart.js]: What is newCart? The name of the file should be updated

File: assets/scripts/newCart.js [hunk 15]
```diff
@@ -0,0 +1,134 @@
...
+import { updateCartCount } from "./updateCartCount.js";
+
+// Variables
+let cartContainer = document.querySelector(".cart-products-container");
+let cartSubtotal = document.querySelector(".cart-total-amout");
+let cartCount = document.querySelector(".cart-btn");
+let cartCount2 = document.querySelector(".cart-btn2");
```
Comment (SabryShawally) [inline assets/scripts/newCart.js]: CartCount2 doesn’t make any sense

Comment (SabryShawally) [reply assets/scripts/newCart.js]: This file seems to be a duplicate of cart.js

**Review by SabryShawally — CHANGES_REQUESTED** [review]
Overall a good start! Please find some comments I added since I didn’t do a full review. Please focus on readability and consistency in your coding style as it would help me as a reader and you later if you want to make changes later in time

File: assets/scripts/home.js (hunk 11 above)
Comment (rihan-a) [reply assets/scripts/home.js]: fixed

File: assets/css/style.css (hunk 1 above)
Comment (rihan-a) [reply assets/css/style.css:75]: Pro stand for product

File: assets/css/style.css (hunk 3 above)
Comment (rihan-a) [reply assets/css/style.css:324]: This is a good idea, I will split them and commit.

**Review by SabryShawally — APPROVED** [review]
LGTM approving for now

**Pull request merged**